src/app/campaign/store/selectors/combined.selectors.ts

```diff
@@ -0,0 +1,42 @@
+import { createSelector } from '@ngrx/store';
+import { selectCampaign, selectCampaignLoaded, selectCampaignError } from './campaign.selectors';
+import { selectAllFlights, selectFlightEntities, selectRoutedFlightId } from './flight.selectors';
+import { CampaignState, FlightState, CampaignFormSave } from '../reducers';
+
+export const selectCampaignWithFlightsForSave = createSelector(
+  selectCampaign,
+  selectAllFlights,
+  (campaign: CampaignState, flights: FlightState[]): CampaignFormSave => ({
+    campaign: campaign.localCampaign,
+    updatedFlights: flights
+      .filter(flight => !flight.softDeleted && flight.changed && flight.remoteFlight)
+      .map(flight => flight.localFlight),
+    createdFlights: flights.filter(flight => !flight.softDeleted && !flight.remoteFlight).map(flight => flight.localFlight),
+    deletedFlights: flights.filter(flight => flight.softDeleted && flight.remoteFlight).map(flight => flight.localFlight)
+  })
+);
+
+export const selectValid = createSelector(
+  selectCampaign,
+  selectAllFlights,
+  (campaignState: CampaignState, flightsState: FlightState[]): boolean =>
+    campaignState.valid && flightsState.every(flight => (flight.valid && !!flight.localFlight.totalGoal) || flight.softDeleted)
+);
+
+export const selectChanged = createSelector(
+  selectCampaign,
+  selectAllFlights,
+  (campaignState: CampaignState, flightsState: FlightState[]): boolean =>
+    campaignState.changed || flightsState.some(flight => flight.changed || flight.softDeleted)
+);
+
+export const selectFlightNotFoundError = createSelector(
+  selectCampaignLoaded,
+  selectRoutedFlightId,
+  selectFlightEntities,
+  (loaded, flightId, flights): string => loaded && flightId && !flights[flightId] && 'Flight Not Found'
+);
+
+export const selectError = createSelector(selectCampaignError, selectFlightNotFoundError, (campaignError: any, flightError: string) => {
```

Comment:
Could stand to do some better parsing off error messages, but we're displaying them now. (Make a change to a flight with a date in the past, get Error 422, see error shown in app)

Reply:
There's also an Augury ticket out there to make the message part nicer - so may want to wait for that to land first.